Add variant generic interfaces introduced in 4.5

csrowell opened the pull request:
Many places introduce this page as "For the list of the variant interfaces in the .NET Framework, see Variance in Generic Interfaces (C#)." so I am thinking it would be nice to include the two new ones introduced in .NET Framework 4.5.

csrowell committed:
Add variant generic interfaces introduced in 4.5

Many places introduce this page as "For the list of the variant interfaces in the .NET Framework, see Variance in Generic Interfaces (C#)." so I am thinking it would be nice to include the two new ones introduced in .NET Framework 4.5

rpetrusha commented on docs/csharp/programming-guide/concepts/covariance-contravariance/variance-in-generic-interfaces.md:
```diff
@@ -6,7 +6,9 @@ ms.assetid: 4828a8f9-48c0-4128-9749-7fcd6bf19a06
-.NET Framework 4 introduced variance support for several existing generic interfaces. Variance support enables implicit conversion of classes that implement these interfaces. The following interfaces are now variant:
+.NET Framework 4 introduced variance support for several existing generic interfaces. Variance support enables implicit conversion of classes that implement these interfaces. 
+
+The following interfaces are now variant:
```
```suggestion
Staring with .NET Framework 4, the following interfaces are variant:
```

rpetrusha commented on docs/csharp/programming-guide/concepts/covariance-contravariance/variance-in-generic-interfaces.md:
```diff
@@ -22,6 +24,12 @@ ms.assetid: 4828a8f9-48c0-4128-9749-7fcd6bf19a06
 
 - <xref:System.IComparable%601> (T is contravariant)
 
+The following variant interfaces were added in .NET Framework 4.5:
```
```suggestion
Starting with .NET Framework 4.5, the following interfaces are variant:
```

rpetrusha commented on docs/csharp/programming-guide/concepts/covariance-contravariance/variance-in-generic-interfaces.md:
````diff
@@ -22,6 +24,12 @@ ms.assetid: 4828a8f9-48c0-4128-9749-7fcd6bf19a06
+
 Covariance permits a method to have a more derived return type than that defined by the generic type parameter of the interface. To illustrate the covariance feature, consider these generic interfaces: `IEnumerable<Object>` and `IEnumerable<String>`. The `IEnumerable<String>` interface does not inherit the `IEnumerable<Object>` interface. However, the `String` type does inherit the `Object` type, ...
 
 ```csharp
````
unrelated to your PR: on line 40, it's VB and not C# that has an `Option Strict On` setting. Could you change this to read "...a compilation error in C# and, if `Option Strict` is on, in Visual Basic."

csrowell replied:
ah, good catch, will do.

rpetrusha commented on docs/csharp/programming-guide/concepts/covariance-contravariance/variance-in-generic-interfaces.md:
```diff
@@ -6,7 +6,9 @@ ms.assetid: 4828a8f9-48c0-4128-9749-7fcd6bf19a06
 
```
line 3 contains ms.date, the date this topic was last updated. Please change it to today's date.

csrowell replied:
Yep, will do.

rpetrusha reviewed (changes requested):
Thanks, @csrowell, for updating this article. I suggested some changes and also added two comments unrelated to your PR.

Commit:
Update docs/csharp/programming-guide/concepts/covariance-contravariance/variance-in-generic-interfaces.md

Co-Authored-By: csrowell <csrowell@users.noreply.github.com>

Commit:
Update docs/csharp/programming-guide/concepts/covariance-contravariance/variance-in-generic-interfaces.md

Co-Authored-By: csrowell <csrowell@users.noreply.github.com>

csrowell committed:
Clarify behavior and update date

Clarified that the `Option Strict` refers to VB only; updated ms.date.

csrowell commented:
I think I addressed all of the comments. I apologize because I am not overfly familiar with this process, but it isn't immediately clear if I need to do anything to move this one forward. So please let me know if there's something else I need to do!

rpetrusha commented:
No, there's nothing you need to do to move this forward, @csrowell. Thank you very much both for the original PR and for making the additional changes, including the ones unrelated to your PR. I'll approve and merge your PR now. The changes should be live on docs.microsoft.com in the next day or two.
